src/core/AutomationPattern.cpp

```diff
@@ -454,37 +454,32 @@ float *AutomationPattern::valuesAfter( const MidiTime & _time ) const
 
 
 
-void AutomationPattern::flipY( int min, int max )
+void AutomationPattern::flipY(int min, int max)
 {
-	timeMap tempMap = m_timeMap;
-	timeMap::const_iterator iterate = m_timeMap.lowerBound(0);
-	float tempValue = 0;
-	float outValueOffset = 0;
-
-	int numPoints = 0;
-
-	// TODO: This loop looks really odd. Is there a particular case where iterate+i+1 != m_timeMap.end()
-	// will be true but iterate + 1 != m_timeMap.end() will be false?
-	for( int i = 0; ( iterate + i + 1 ) != m_timeMap.end() && ( iterate + i ) != m_timeMap.end() ; i++)
+	timeMap::const_iterator it = m_timeMap.lowerBound(0);
+	if (it == m_timeMap.end() )
```

Comment:
We can also merge the 4 lines of this if into a single line since it's so short.

```suggestion
	if (it == m_timeMap.end())
```